tortoise/filters.py

```diff
@@ -381,15 +406,73 @@ def get_json_filter_operator(
     return key_parts, filter_value, operator_
 
 
+def get_array_filter(
+    field_name: str, source_field: str, field: ArrayField
+) -> dict[str, FilterInfoDict]:
+    return {
+        field_name: {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": operator.eq,
+            "value_encoder": array_encoder,
+        },
+        f"{field_name}__not": {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": not_equal,
+            "value_encoder": array_encoder,
+        },
+        f"{field_name}__isnull": {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": is_null,
+            "value_encoder": bool_encoder,
+        },
+        f"{field_name}__not_isnull": {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": not_null,
+            "value_encoder": bool_encoder,
+        },
+        f"{field_name}__contains": {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": array_contains,
+            "value_encoder": array_encoder,
+        },
+        f"{field_name}__contained_by": {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": array_contained_by,
+            "value_encoder": array_encoder,
+        },
+        f"{field_name}__overlap": {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": array_overlap,
+            "value_encoder": array_encoder,
+        },
+        f"{field_name}__len": {
+            "field": field_name,
+            "source_field": source_field,
+            "operator": array_length,
+            "value_encoder": int_encoder,
+        },
+    }
+
+
 def get_filters_for_field(
     field_name: str, field: Optional[Field], source_field: str
 ) -> dict[str, FilterInfoDict]:
-    if isinstance(field, ManyToManyFieldInstance):
-        return get_m2m_filters(field_name, field)
-    if isinstance(field, BackwardFKRelation):
-        return get_backward_fk_filters(field_name, field)
-    if isinstance(field, JSONField):
-        return get_json_filter(field_name, source_field)
+    if field:
```

Comment:
I think that `if field is not None` is better.

Reply:
updated